src/material/button/button.ts

```diff
@@ -31,18 +37,87 @@ import {MAT_BUTTON_HOST, MatButtonBase} from './button-base';
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class MatButton extends MatButtonBase {}
+export class MatButton extends MatButtonBase {
+  /** Appearance of the button. */
+  @Input('matButton')
+  get appearance(): MatButtonAppearance | null {
+    return this._appearance;
+  }
+  set appearance(value: MatButtonAppearance | '') {
+    // Allow empty string so users can do `<button matButton></button>`
+    // without having to write out `="text"` every time.
+    this.setAppearance(value || this._config?.defaultAppearance || 'text');
+  }
+  private _appearance: MatButtonAppearance | null = null;
+
+  constructor(...args: unknown[]);
+
+  constructor() {
+    super();
+    const element = this._elementRef.nativeElement;
+    const inferredAppearance = _inferAppearance(element);
+
+    // This class is common across all the appearances so we add it ahead of time.
+    element.classList.add('mdc-button');
```

Comment:
would be easier to just make this a host binding?

Reply:
I was trying to match the behavior of what we do at the moment. Moving it into a host binding can change the order in which the class is applied which can break internal clients and I was trying to minimize the amount of breakages. I can try moving it in a follow-up. We'll have to shuffle around the `host` object a bit anyways if we want to get type checking support for it.

Reply:
Ah okay that makes sense